### DISCUSSION TITLE
STT+TTS Request: VK Cloud (russian)

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#145] [Dev]
We should create a OPM plugin based on https://github.com/MycroftAI/mycroft-core/pull/3024 we also need to create a yandex plugin and support it in `Factory.create()`, it used to be present in mycroft-core and was removed in https://github.com/OpenVoiceOS/ovos-core/pull/1

For proper language support we should aim to have plugins for as many languages as possible, specially those who we know have an active community. 

This does not necessarily bring extra compatibility right now, but future proofs ovos-core while enabling more use cases:
- we can be sure OPM plugins always work in mycrof-core , ovos-core and standalone (no mycroft needed!)
- we do not know when a mycroft-core plugin will break compatibility with ovos-core (eg, by depending on a new import we didnt sync yet or internal refactoring of the TTS base class)

**Comment by krisgesling** [comment]
I may be misinterpreting this - it seems to read as if OVOS is going to create forks of every plugin. 

So there will be a `mycroft-plugin-vk-cloud` and an `ovos-plugin-vk-cloud` - is that right?

**Comment by JarbasAl** [comment]
> I may be misinterpreting this - it seems to read as if OVOS is going to create forks of every plugin.
> 
> So there will be a `mycroft-plugin-vk-cloud` and an `ovos-plugin-vk-cloud` - is that right?

We haven't really discussed this, these latest issues were also meant as conversation starters

At minimum i think we want plugins for every engine supported by mycroft-core, this is because ovos-core uses OPM directly and support was dropped for a few engines, we want any mycroft.conf in the wild to be fully compatible so we need to restore support for the removed engines

We have also been discussing onboarding, and language selection often comes into the discussion, to offer full language support we need male/female offline/online combinations for each language

Any plugin that requires importing `mycroft` can only be used in mycroft, ovos plugins can be used standalone. So in the end i think the decisive factor is what non-mycroft usage do we want to support?

Personally my motivations for these issues was the hivemind voice satellite, I'd like to provide at least 1 plugin for each supported language (supported meaning core + LF). 

The local backend also benefits a lot from this language support, but only in regards to STT

Forking every plugin is not a realistic option, but forking the important ones to ensure minimal lang support is doable. 

Our other option would be to convince people to use OPM instead of mycroft in plugins we care about

We just published an initial list of all know OPM plugins here https://github.com/OpenVoiceOS/awesome-ovos-plugins

**Comment by ChanceNCounter** [comment]
@krisgesling as a rule, we lack structure, so OVOS forks what somebody decides to fork. In practice, however, we are all off the clock, unless Neon or Chatterbox happen to need something from one of our libs during business hours.

So it really has to be worth the effort.

There are two circumstances under which we have to fork or dupe a repo:

* Upstream won't run downstream
* Upstream doesn't have or want features we need or want

Otherwise it's an unbelievable waste of dev time to fork
